infra/modules/providers/azure/function-app/main.tf

```diff
@@ -1,89 +1,89 @@
+data "azurerm_client_config" "current" {}
+
 locals {
-  access_restriction_description = "blocking public traffic to function app"
+  fn_names                       = keys(var.fn_app_config)
+  fn_configs                     = values(var.fn_app_config)
   access_restriction_name        = "vnet_restriction"
-  function_app_names             = keys(var.function_app_config)
-
-  // Add app setting only if instrumentation key is set.
-  insights_settings = var.instrumentation_key != "" ? {
-    APPINSIGHTS_INSTRUMENTATIONKEY = var.instrumentation_key
-  } : {}
-
-  // Only Apply Docker App Settings if all values for URL, UserName and Password are given.
-  docker_settings = length(compact([var.docker_registry_server_username, var.docker_registry_server_password, var.docker_registry_server_url])) == 3 ? {
-    "DOCKER_REGISTRY_SERVER_URL" : format("https://%s", var.docker_registry_server_url)
-    "DOCKER_REGISTRY_SERVER_USERNAME" : var.docker_registry_server_username
-    "DOCKER_REGISTRY_SERVER_PASSWORD" : var.docker_registry_server_password
-  } : {}
+  access_restriction_description = "blocking public traffic to function app"
 
-  app_settings = merge(
-    local.insights_settings,
-    local.docker_settings,
-  )
 
   app_linux_fx_versions = [
-    for config in values(var.function_app_config) :
-    // Without specifyin a `linux_fx_version` the webapp created by the `azurerm_function_app` resource
-    // will be a non-container webapp.
-    //
-    // The value of "DOCKER" is a stand-in value that can be used to force the webapp created to be
-    // container compatible without explicitly specifying the image that the app should run.
-    config.image == "" ? "DOCKER" : format("DOCKER|%s/%s", var.docker_registry_server_url, config.image)
+    for config in values(var.fn_app_config) :
+    config.image == null ? "" : format("DOCKER|%s/%s", var.docker_registry_server_url, config.image)
+  ]
+
+  app_deployment_config = [
+    for config in values(var.fn_app_config) :
```

Comment:
Suggestion: Replace with `for config in local.fn_configs `. I'm suggesting because line 34 proves that local variables can references other local variables.